### PR TITLE
Add Golang support to ImageBuilder

### PR DESCRIPTION
**Pull request by sharonlinTO**
Work Item: [Task 10401248](https://msazure.visualstudio.com/Antares/_workitems/edit/10401248)

Behavior: Once a Docker image is deployed to container web app, this will be the expected output behavior:
image

### CODE REVIEW

**Comment by nertim** [comment]
@sanchitmehta - One of the things you will see in this PR is that we have the hostingstart binary being pushed up. Ideally, I would prefer that the CI.yml generate the hostingstart binary.. What do you think? if so, what is the best way to test/playaround with that

File: GenerateDockerFiles/go/alpine/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,68 @@
+## First step: Build Golang binary
+FROM golang:1.16-alpine3.14 as BUILDER
+RUN mkdir /build
+ADD hostingstart.go /build/
+WORKDIR /build
+RUN go mod init hostingstart
+RUN CGO_ENABLED=0 GOOS=linux go build -a -o hostingstart .
+
```
Comment (nertim) [inline GenerateDockerFiles/go/alpine/Dockerfile:8]: remove extra newline

File: GenerateDockerFiles/go/alpine/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+## Second step: Copy executable
+FROM alpine:3.14
+LABEL maintainer="Azure App Services Container Images <appsvc-images@microsoft.com>"
+ENV HOME_SITE "/home/site/wwwroot"
+ENV APP_PATH "/home/site/wwwroot"
+
+# Setup SSH and activate virtual env on ssh into containers
+# NOTE: dnsutils is not available in apk
```
Comment (nertim) [inline GenerateDockerFiles/go/alpine/Dockerfile:17]: @sanchitmehta - thoughts on this? basically is dnsutils needed? if so, we will need to figure out an alternative way of getting it.

File: GenerateDockerFiles/go/generateDockerfiles.sh [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+# Current Working Dir
+declare -r DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null && pwd )"
+
+# Directory for Generated Docker Files
+declare -r STACK_NAME="go"
+declare -r SYSTEM_ARTIFACTS_DIR="$1"
+declare -r CONFIG_DIR="$5"                                         # ${Current_Repo}/Config
+declare -r APP_SVC_REPO_DIR="$SYSTEM_ARTIFACTS_DIR/$STACK_NAME/GitRepo"
+
```
Comment (nertim) [inline GenerateDockerFiles/go/generateDockerfiles.sh:18]: only one newline needed.

File: GenerateDockerFiles/go/alpine/Dockerfile (hunk 2 above)
Comment (sanchitmehta) [reply GenerateDockerFiles/go/alpine/Dockerfile:17]: This is typically for self diagnosis for the users. We could instead use the `bind-tools`: https://pkgs.alpinelinux.org/package/edge/main/x86/bind-tools

**Review by nertim — APPROVED** [review]

**Pull request merged**